zkvm/src/signature/musig.rs

```diff
@@ -0,0 +1,176 @@
+#![allow(non_snake_case)]
+
+use crate::errors::VMError;
+use crate::signature::VerificationKey;
+use crate::transcript::TranscriptProtocol;
+use curve25519_dalek::constants::RISTRETTO_BASEPOINT_POINT;
+use curve25519_dalek::ristretto::RistrettoPoint;
+use curve25519_dalek::scalar::Scalar;
+use merlin::Transcript;
+
+#[derive(Clone)]
+pub struct PubKey(pub RistrettoPoint);
+
+#[derive(Clone)]
+pub struct PrivKey(pub Scalar);
+
+#[derive(Debug, Clone)]
+pub struct Signature {
+    pub s: Scalar,
+    pub R: RistrettoPoint,
+}
+
+#[derive(Clone)]
+pub struct Message(pub Vec<u8>);
```

Comment:
do we need this type?

Reply:
removed - it looks fine. I wasn't sure how much abstraction / wrapping we wanted so I originally leaned towards more wrappers.